Switch to webpack via wordpress-scripts and simplify build

jom opened the pull request:
### Changes proposed in this Pull Request

* Standardizes `npm` scripts with core Sensei.
* Switches to `wordpress-scripts` (webpack) for building assets.
* Removes gulpfile.js and dependencies.
* Breaking Dev Change: No longer using `build` directory.

### Testing instructions

* Run `npm install` and `npm run build` and test generated `sensei-content-drip.zip` file.

jom committed:
Remove built and unused assets

jom committed:
Rename jquery-ui.css to scss for minimization

jom committed:
Point assets to new `dist` directory

jom committed:
Introduce new build system using wordpress-scripts and webpack

jom committed:
Remove gulpfile

jom committed:
Pin dependency

jom committed:
Set file paths to const

jom committed:
Add editor config

gkaragia commented on assets/css/admin-lesson.css:
```diff
@@ -1,19 +0,0 @@
-/**
```
This file is probably needed as the UI has an issue:
![Screenshot 2020-09-02 at 1 30 47 PM](https://user-images.githubusercontent.com/53191348/91970629-9ef7d580-ed20-11ea-88be-3411290d9cf9.png)

jom replied:
@gkaragia It is still there, just moved to `admin-lesson.scss`. Git didn't track it as a rename for some reason. I think this is a bug in the style. When I inspect the interval number field, I can see the style applied from the generated `admin-lesson.scss`.

jom replied:
A-ha! I see what happened. There was already an `admin-lesson.scss` file, however the css file was getting updates that the SCSS file wasn't. I'll bring the changes over to the scss file.

gikaragia reviewed (commented):
There is probably some CSS missing.

jom committed:
Update `@wordpress/scripts` to 12.2.0

jom committed:
Port over unit amount fix from CSS to the SCSS file

https://github.com/woocommerce/sensei-content-drip/commit/87762d4a12759dc53d4d4b3da431912b4b1cda3b was made to CSS not SCSS

gikaragia reviewed (approved):
Issue is fixed now! I know it is an existing issue, but is it possible to add some space between the button and the input field to make it conistent with the other options?
![Screenshot 2020-09-08 at 12 40 44 PM](https://user-images.githubusercontent.com/53191348/92460484-b0355c00-f1d0-11ea-8ebe-7e021d85443a.png)

jom committed:
Give proper padding to drop options on lesson editor

jom commented:
@gkaragia We were surrounding `<div>` with `<p>`. I set the style for the `<div>`s and seems to look better. https://github.com/woocommerce/sensei-content-drip/pull/201/commits/7381da45752d90a63dfb9ac46c32d63ed55cf8c8

gikaragia reviewed (approved):
LGTM!
